tests/attr/test_lrp.py

```diff
@@ -0,0 +1,314 @@
+        lrp = LRP(model)
+        self.assertRaises(TypeError, lrp.attribute, inputs)
+
+    def test_lrp_basic_attributions(self):
+        model, inputs = _get_basic_config()
+        logits = model(inputs)
+        _, classIndex = torch.max(logits, 1)
+        lrp = LRP(model)
+        relevance, delta = lrp.attribute(
+            inputs, classIndex.item(), return_convergence_delta=True
+        )
+        self.assertEqual(delta.item(), 0)
+        self.assertEqual(relevance.shape, inputs.shape)
+
+    def test_lrp_simple_attributions(self):
+        model, inputs = _get_simple_model()
+        model.eval()
+        model.linear.rule = EpsilonRule()
+        model.linear2.rule = EpsilonRule()
+        lrp = LRP(model)
+        relevance = lrp.attribute(inputs)
+        assertTensorAlmostEqual(
+            self, relevance, torch.tensor([18 / 108, 36 / 108, 54 / 108])
+        )
+
+    def test_lrp_simple_attributions_batch(self):
+        model, inputs = _get_simple_model()
+        model.eval()
+        model.linear.rule = EpsilonRule()
+        model.linear2.rule = EpsilonRule()
+        lrp = LRP(model)
+        inputs = torch.stack((inputs, 3 * inputs))
+        relevance, delta = lrp.attribute(inputs, return_convergence_delta=True)
+        self.assertEqual(relevance.shape, inputs.shape)
+        self.assertEqual(delta.shape[0], inputs.shape[0])
+
+    def test_lrp_simple_repeat_attributions(self):
+        model, inputs = _get_simple_model()
+        model.eval()
+        model.linear.rule = GammaRule()
+        model.linear2.rule = Alpha1_Beta0_Rule()
+        output = model(inputs)
+        lrp = LRP(model)
+        _ = lrp.attribute(inputs)
+        output_after = model(inputs)
+        assertTensorAlmostEqual(self, output, output_after)
+
+    def test_lrp_simple_inplaceReLU(self):
+        model_default, inputs = _get_simple_model()
+        model_inplace, _ = _get_simple_model(inplace=True)
+        for model in [model_default, model_inplace]:
+            model.eval()
+            model.linear.rule = EpsilonRule()
+            model.linear2.rule = EpsilonRule()
+        lrp_default = LRP(model_default)
+        lrp_inplace = LRP(model_inplace)
+        relevance_default = lrp_default.attribute(inputs)
+        relevance_inplace = lrp_inplace.attribute(inputs)
+        assertTensorAlmostEqual(self, relevance_default, relevance_inplace)
+
+    def test_lrp_simple_tanh(self):
+        class Model(nn.Module):
+            def __init__(self):
+                super(Model, self).__init__()
+                self.linear = nn.Linear(3, 3, bias=False)
+                self.linear.weight.data.fill_(0.1)
+                self.tanh = torch.nn.Tanh()
+                self.linear2 = nn.Linear(3, 1, bias=False)
+                self.linear2.weight.data.fill_(0.1)
+
+            def forward(self, x):
+                return self.linear2(self.tanh(self.linear(x)))
+
+        model = Model()
+        _, inputs = _get_simple_model()
```

Comment:
it's a bit confusing that we have to call `_get_simple_model ` to get the inputs. Could wee, please, define the inputs here ?

Reply:
Yes, I added the inputs here